process/interceptors/common.go

```diff
@@ -20,23 +23,30 @@ func preProcessMesage(
 	if message.Data() == nil {
 		return process.ErrNilDataToProcess
 	}
-	err := antifloodHandler.CanProcessMessage(message, fromConnectedPeer)
-	if err != nil {
-		return err
-	}
-	err = antifloodHandler.CanProcessMessagesOnTopic(fromConnectedPeer, topic, 1, uint64(len(message.Data())), message.SeqNo())
-	if err != nil {
-		return err
-	}
 
-	if !throttler.CanProcess() {
-		return process.ErrSystemBusy
+	if !isMessageFromSelfToSelf(fromConnectedPeer, message, id) {
+		err := antifloodHandler.CanProcessMessage(message, fromConnectedPeer)
+		if err != nil {
+			return err
+		}
+		err = antifloodHandler.CanProcessMessagesOnTopic(fromConnectedPeer, topic, 1, uint64(len(message.Data())), message.SeqNo())
+		if err != nil {
+			return err
+		}
+
+		if !throttler.CanProcess() {
+			return process.ErrSystemBusy
+		}
 	}
 
 	throttler.StartProcessing()
 	return nil
 }
 
+func isMessageFromSelfToSelf(fromConnectedPeer core.PeerID, message p2p.MessageP2P, id core.PeerID) bool {
+	return bytes.Equal(message.Signature(), id.Bytes()) && bytes.Equal(message.From(), id.Bytes()) && fromConnectedPeer == id
```

Comment:
this is a convention since the self-to-self messages are hardcoded to have the signature == from. All 4 values need to be equal in order to be 100% sure there are no connected malicious peers that send some fake messages in order to fool the interceptor.